### ISSUE TITLE
It works on Ubuntu but does not work on the WIndow.

### ISSUE THREAD

**Issue by kyunghwan** [AlexeyAB/darknet#148]
Hi
I install darknet window version.
(I checked operation using yolo.cfg, yolo.weights, coco.data.)

I used darknet ubuntu version to train my own data.
I checked operation my weights in ubuntu

But, it does't work at windows version
Please how to solve this problem?

**Comment by VanitarNordic** [comment]
Please describe more or add some screenshots of what problem do you have in details

**Comment by AlexeyAB** [comment]
@kyunghwan 

* Show you .cfg-file and .data-file
* Is your .cfg-file based on `yolo-voc.cfg` or on `yolo-voc.2.0.cfg`?
* Try to compile this fork on Linux, can you use your own model on this fork successfully?

**Comment by kyunghwan** [comment]
Thank you everyone.
@AlexeyAB 
I used cfgfile based on yolo-voc.2.0.cfg. (Does this affect the results differently?)
and I used pjreddie/darknet newest version.

I will try using this fork version.

**Comment by VanitarNordic** [comment]
@AlexeyAB 

What is the actual difference between `yolo-voc.cfg` and `yolo-voc.2.0.cfg`?
V2.0 tends to be better and we should train and test using it?

**Comment by AlexeyAB** [comment]
@kyunghwan @VanitarNordic 
This fork supports on detection task any cfg. But this fork supports for training only `yolo-voc.2.0.cfg`.
`yolo-voc.cfg` a little bit smaller and a little bit faster.

One of diff is addition convolutional layer with size=1, that convolve 26x26x512 to 26x26x64, i.e. it decreases size of tensor before reorg-layer: https://github.com/AlexeyAB/darknet/blob/e4323274f2ab75a6813e259f852ee3d8cf335fa5/cfg/yolo-voc.cfg#L211

This affects the detection of small objects, but I have not tested whether it improves or worsens detection.

**Comment by VanitarNordic** [comment]
I had tested the `yolo-voc.cfg` before. regarding small object I can say if they are included in the training set in various scales, then it will detect them.

But in the next few days I will train and test again with this new `yolo-voc.2.0.cfg` and tell you the results or any differences. if they have added some layers then theoretically it should improve the accuracy, but minor changes in speed because the modification is not massive and it seems like a model tuning.

**Comment by VanitarNordic** [comment]
> This fork supports on detection task any cfg. But this fork supports for training only ...

I think you made a typo mistake. You mean this final fork just should be trained with `yolo-voc.2.0.cfg`, but can be tested with this cfg and also others?

**Comment by AlexeyAB** [comment]
@VanitarNordic Yes.

**Comment by anmikh** [comment]
@kyunghwan  When you say "But, it doesn't work at windows version" - do you mean it doesn't detect objects at all or detections are incorrect? 

If it doesn't detect objects at all, you can try to modify file `parser.c` in this windows fork.  I've met the same problem too (I'm not sure that it's exactly the same) but  these modifications helped me:

 replace this [line ](https://github.com/AlexeyAB/darknet/blob/master/src/parser.c#L1026) of code:

`    fread(net->seen, sizeof(int), 1, fp);`

with the next [block ](https://github.com/pjreddie/darknet/blob/1e729804f61c8627eb257fba8b83f74e04945db7/src/parser.c#L1109-L1115): 
```
    if ((major*10 + minor) >= 2){
        fread(net->seen, sizeof(size_t), 1, fp);
    } else {
        int iseen = 0;
        fread(&iseen, sizeof(int), 1, fp);
        *net->seen = iseen;
    }
```

**Comment by VanitarNordic** [comment]
@AlexeyAB 

I did training with the latest fork and `yolo-voc.2.0.cfg`. The used dataset was exactly identical with the older forks:

Validation with yolo-voc.2.0.cfg : `IoU=69%    Recall=86%`
Validation with yolo-voc.cfg : `IoU=69%    Recall=86%`
Validation in older forks (after final random function modification): `IoU=75%    Recal=94%`

**Comment by AlexeyAB** [comment]
@VanitarNordic Thank you! What do you mean about older and latest fork, which repo & commit?

**Comment by VanitarNordic** [comment]
@AlexeyAB 

I don't remember the repo but I mean the last commit where we solved the random function issue with rand_s. I had kept a copy of that repo in my hard disk. 

I'm continuously testing the last repo. it might be a minor issue somewhere in the code. if this new .cfg has introduced by the authors, then the problem "might be" from somewhere else such as the code.

**Comment by VanitarNordic** [comment]
@AlexeyAB 

I trained the last repo with old `yolo-voc.cfg` and also made a validation with it: `IoU=70.7% Recall=86.5%`

**Comment by VanitarNordic** [comment]
@AlexeyAB 

I tried again and I can not get higher values with the latest repo. Could be something related to the OpenCV versions? because in the past it was fixed to 2.4.9.
if not, please have a look at what has changed from past. if you inform me, then I can have a look because I have focus on this right now.

**Comment by AlexeyAB** [comment]
@VanitarNordic Are you about my repo? https://github.com/AlexeyAB/darknet

5 Apr I changed threshold in `validate_detector_recall()` from ` 0.001;` to `0.2;`: https://github.com/AlexeyAB/darknet/blame/master/src/detector.c#L397
Because the value of 0.2 is closer to the real threshold, which is used for detection (default 0.24).

So now you get more real IoU than earlier.

**Comment by VanitarNordic** [comment]
@AlexeyAB 

Yes, I am talking about your repo. changing that value changes `RPs/Img`. I don't think that threshold is the same as detection threshold because 0.001 is very low (it is 0.1%) and if we use it in detection then the screen will be full of bounding boxes.

**Comment by AlexeyAB** [comment]
@VanitarNordic Recall threshold is the same as detection threshold.
Yolo calculates only the best of IoU instead of average IoU, so if we set threshold 0.001, then yolo will remove any bad bounding boxes exclude one the best: https://github.com/AlexeyAB/darknet/blob/489207111f1cdf0d7e3bcfc2410a92121c58f3cd/src/detector.c#L435

**Comment by VanitarNordic** [comment]
@AlexeyAB 

I'm not quite sure because that might be the IoU value when the model trains itself, I mean the `Avg IoU` in console during training, 
if your comment is right, then actually we are not fine-tuning the model, because results are much worse than the initial thoughts when we fine-tune a model. I mean it must be much better. I think the provided weights is a initial weight used for training.

**Comment by AlexeyAB** [comment]
@kyunghwan I added some fixes. Just try to update your code from this repository and try to use your model.

**Comment by VanitarNordic** [comment]
@AlexeyAB 

I'll do that and tell you the results. But I don't think the authors intentionally used a low number to produce non-realistic results and publish a paper.

**Comment by AlexeyAB** [comment]
@VanitarNordic Authors didn't publish IoU results anywhere. This is a purely internal indicator for own needs.

**Comment by VanitarNordic** [comment]
@AlexeyAB 

You don't change your mind no problem :-) but I got worse detection using that (older repo detecting better). I'll try your new code to see what will happen.

**Comment by AlexeyAB** [comment]
@VanitarNordic Just set `float thresh = .001;` and try: https://github.com/AlexeyAB/darknet/blob/489207111f1cdf0d7e3bcfc2410a92121c58f3cd/src/detector.c#L397

**Comment by VanitarNordic** [comment]
@AlexeyAB 

I tested the last commit with `thresh = 0.2` and trained using the `yolo-voc.2.0.cfg`
The results are Good. although IoU shows few percent lower but detection is Okay.

I'll try with the older `.cfg` and check if I can get better results.
